src/test/java/com/iexec/common/docker/client/DockerClientInstanceTests.java

```diff
@@ -411,6 +413,21 @@ public void shouldPullImage() {
         dockerClientInstance.removeImage(ALPINE_LATEST);
     }
 
+    @Test
+    public void shouldPullImageWithExplicitTimeout() {
+        assertThat(dockerClientInstance.pullImage(ALPINE_LATEST, Duration.of(1, ChronoUnit.MINUTES))).isTrue();
+        assertThat(dockerClientInstance.isImagePresent(ALPINE_LATEST)).isTrue();
+        dockerClientInstance.removeImage(ALPINE_LATEST);
```

Comment:
I think the image should be removed at the beginning of test.
Nothing guarantees it exists or not (could be pulled from integration tests) which could lead to discrepancies in executions.

Reply:
I tend to agree with you. However, they are all pulled at the beginning of each test: https://github.com/iExecBlockchainComputing/iexec-common/blob/40dd9019f961e95a853ade189b5de5671e104039/src/test/java/com/iexec/common/docker/client/DockerClientInstanceTests.java#L99
I'm not really sure why.

Reply:
I disagree.
The method is annotated with `@BeforeAll`.
Images are pulled once before the execution of the first test.

Reply:
Good catch! Confusion with the other `@BeforeEach` annotated method.

Reply:
I don't know how, but I think we need to guarantee 2 things:
* Tests start with a predictable state
* The test implementation (timeout values) for tests are in such a way that we guarantee the tests will succeed (we want to avoid infrastructure side effects because other processes run on test machines)
That's why I suggested to have a big timeout because we don't want to measure the speed of the download, we want to assure that if the download happens (even if slow), the returned result is `true`.

Reply:
https://hub.docker.com/_/alpine?tab=tags
![image](https://user-images.githubusercontent.com/33313130/160440914-d2e905c5-e24d-454f-82f3-8942eb33eb5c.png)
I'm fine with 1min, but do you prefer 2min or else?

Reply:
I think we need a security factor for the test, just to avoid false positive and side-effects due to the infrastructure being used to do something else at the same time.
I would even put a timeout around 3 minutes, just in case.

Reply:
Ok, updated to 3 minutes. See https://github.com/iExecBlockchainComputing/iexec-common/pull/315/commits/20d2d531bf6f58fd837eb09c07039654f266b3ba.

Reply:
Thank you for the update